src/auth/Crypto.cc

```diff
@@ -135,9 +135,11 @@ static int nss_aes_operation(
 			     const bufferlist& in, bufferlist& out,
 			     std::string *error)
 {
-  // sample source said this has to be at least size of input + 8,
-  // but i see 15 still fail with SEC_ERROR_OUTPUT_LEN
-  bufferptr out_tmp(in.length()+16);
+  // CK_RC2_CBC_PARAMS is the struct used for IV param with CKM_AES_CBC_PAD
+  // we could get it from param->len. but the param is now ectx->param, and
+  // it's internals in NSS.
+  static constexpr unsigned iv_size = sizeof(CK_RC2_CBC_PARAMS);
```

Comment:
I find it surprising that CK_RC2_CBC_PARAMS is used for AES.

Reply:
let re-read the code. hold on.

Reply:
@branch-predictor thank you! i was wrong. fixed and repushed again.